pallets/common/src/lib.rs

```diff
@@ -143,3 +147,27 @@ pub fn with_each_transaction<A>(
 ) -> DispatchResult {
     with_transaction(|| iter.into_iter().try_for_each(tx))
 }
+
+/// Either a block number, or nothing.
+#[derive(Copy, Clone, PartialEq, Eq, Encode, Decode, Debug)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum MaybeBlock<BlockNumber> {
```

Comment:
For my education, why this data structure rather than an Option?

Reply:
Originally, it was an `Option<T::BlockNumber>`. However, substrate does funny things with that, stripping the `Option<_>` layer from the genesis config, so you cannot set it to `None`. So we tried `use Option as Maybe;` and that worked, but we (me & Mudit) figured that `MaybeBlock` is a tad clearer.